gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java

```diff
@@ -61,16 +64,17 @@ public List<EndpointGroup> validateAndSanitize(List<EndpointGroup> endpointGroup
                         .getEndpoints()
                         .forEach(
                             endpoint -> {
-                                validateName(endpoint.getName());
+                                validateUniqueEndpointName(endpoint.getName(), names);
```

Comment:
maybe I missed something but according to the README.md, user can select either a `endpointGroup:` or a `endpointName:` in their configuration or attribute.

Here, we validate the uniqueness of the endpoint inside its group. 

Should we assume that if two groups contains each an endpoint with the same name, only the first one will be selected ?

If it's the case, maybe it need an extra explication in the README.md or doc ?
wdyt ?

Reply:
No it validates uniqueness across all the groups. Basically a name (group or endpoint) must be unique.
So, you can't have a group containing an endpoint `my-endpoint` and another group with and endpoint  `my-endpoint`.

If I'm not wrong, I've just reimplemented the same constraints that are applied on v2 definition.

Reply:
my bad, I did not see the set of `names`